### ISSUE TITLE
moving to "modern Python" to-do list

### ISSUE THREAD

**Issue by NickleDave** [nilomr/xeno-canto-py#1]
this is the "to-do" list issue for things along the lines of moving to "modern Python".  
None of this (with the exception of the first PR from you @nilomr) will actually change functionality, but it will help us get organized and stay organized.

After working through this we can take stock and think about how to re-factor -- will start another "discussion" issue with thoughts on re-factoring, including notes from meeting

in general let's follow git workflow (https://www.asmeurer.com/git-workflow/) -- @nilomr as we discussed you could make PRs and then request me to review. Also suggest we follow standard practices for writing commit messages as described here: https://cbea.ms/git-commit/ (Limit subject line to ~50 characters, capitalized, imperative; if we need to capture more context, add a line break then write more)

I would suggest doing things in roughly this order:
- [x] just close PR on ntivirikin repo
- [x] instead make a PR from that branch on **this** repo, then go ahead and merge in those changes
- [x] after merging PR: rename `master` branch to `main`: https://github.com/github/renaming
- [x] also after merging PR, add a docs/changelog.md file and document your initial PR: https://keepachangelog.com/en/1.0.0/
  - just use this simple format for now? https://github.com/vocalpy/crowsetta/blob/main/doc/CHANGELOG.md 
    - I like to include links to the PRs and issues as shown there
  - some projects get clever (e.g. with https://github.com/twisted/towncrier) but that's probably a bit much for our needs at the moment
- [x] move from setup.py -> pyproject.toml
  - a simple way to start would be to use `flit`
  - I prefer this over `poetry` because `poetry` is opinionated in ways that can be annoying later
  - would suggest following an example file: https://github.com/vocalpy/crowsetta/blob/main/pyproject.toml
  - `flit` has helpful docs on PEP 621-style metadata: https://flit.pypa.io/en/latest/pyproject_toml.html?highlight=module#new-style-metadata
- [x] move tests into a tests/ sub-dir
- [x] move xenocanto.py into a src sub-folder
- [x] use `pytest`; add it as a dependency in pyproject.toml 
  - I add it as an `optional-dependency`  (see example pyproject.toml above) which I recently learned some people think is a no-no but I haven't had an issue with it yet; suggest we just go with that for now
- [x] clean up README; first thing should not be dev install
  - I can help write this but basically all the sections + status badges that are here: https://github.com/NickleDave/evfuncs/blob/main/README.md
  - in other words, what writethedocs suggests: https://www.writethedocs.org/guide/writing/beginners-guide-to-docs/#readme
- [x] add noxfile.py for dev
  - again probably just cargo-cult copy on of my files for now: https://github.com/vocalpy/crowsetta/blob/main/noxfile.py
  - unless you already have standard noxfile sessions you prefer, happy to discuss

**Comment by NickleDave** [comment]
@nilomr I know this is a lot to dump into a single issue -- definitely happy to help do more than review PRs, just let me know

**Comment by nilomr** [comment]
See PR #4

**Notes**: 
1. I am calling the existing `unittest` suite using `pytest` in the nox file for now - not worth rewriting for now given major API changes on the horizon.
2. `xenocanto.py` is now under `src/xenocanto`; this assumes that we will not maintain `xenocanto.py` and will have separate modules instead, à la `from xenocanto.download import metadata`.
3. Haven't tested functionality yet [tests: 1 failed, 7 passed, did not look into it]. Possibly not worth doing it before some major reorganisation and rewriting.
